diff --git a/xlsx.mjs b/xlsx.mjs
index 3e61a59..108c6b0 100644
--- a/xlsx.mjs
+++ b/xlsx.mjs
@@ -3920,9 +3920,8 @@ var unescapexml/*:StringConv*/ = /*#__PURE__*/(function() {
 	/* 22.4.2.4 bstr (Basic String) */
 	var encregex = /&(?:quot|apos|gt|lt|amp|#x?([\da-fA-F]+));/ig, coderegex = /_x([\da-fA-F]{4})_/ig;
 	function raw_unescapexml(text/*:string*/)/*:string*/ {
-		var s = text + '', i = s.indexOf("<![CDATA[");
-		if(i == -1) return s.replace(encregex, function($$, $1) { return encodings[$$]||String.fromCharCode(parseInt($1,$$.indexOf("x")>-1?16:10))||$$; }).replace(coderegex,function(m,c) {return String.fromCharCode(parseInt(c,16));});
-		var j = s.indexOf("]]>");
+		var s = text + '', i = s.indexOf("<![CDATA["), j = i == -1 ? -1 : s.indexOf("]]>", i+9);
+		if(i == -1 || j == -1) return s.replace(encregex, function($$, $1) { return encodings[$$]||String.fromCharCode(parseInt($1,$$.indexOf("x")>-1?16:10))||$$; }).replace(coderegex,function(m,c) {return String.fromCharCode(parseInt(c,16));});
 		return raw_unescapexml(s.slice(0, i)) + s.slice(i+9,j) + raw_unescapexml(s.slice(j+3));
 	}
 	return function unescapexml(text/*:string*/, xlsx/*:boolean*/) {
