diff --git a/xlsx.mjs b/xlsx.mjs
index 108c6b0..262427c 100644
--- a/xlsx.mjs
+++ b/xlsx.mjs
@@ -3921,7 +3921,12 @@ var unescapexml/*:StringConv*/ = /*#__PURE__*/(function() {
 	var encregex = /&(?:quot|apos|gt|lt|amp|#x?([\da-fA-F]+));/ig, coderegex = /_x([\da-fA-F]{4})_/ig;
 	function raw_unescapexml(text/*:string*/)/*:string*/ {
 		var s = text + '', i = s.indexOf("<![CDATA["), j = i == -1 ? -1 : s.indexOf("]]>", i+9);
-		if(i == -1 || j == -1) return s.replace(encregex, function($$, $1) { return encodings[$$]||String.fromCharCode(parseInt($1,$$.indexOf("x")>-1?16:10))||$$; }).replace(coderegex,function(m,c) {return String.fromCharCode(parseInt(c,16));});
+		if(i == -1 || j == -1) return s.replace(encregex, function($$, $1) {
+			if(encodings[$$]) return encodings[$$];
+			var code = parseInt($1, $$.toLowerCase().indexOf("x") > -1 ? 16 : 10);
+			if(!Number.isInteger(code) || code < 0 || code > 0x10FFFF) return $$;
+			return String.fromCodePoint(code);
+		}).replace(coderegex,function(m,c) {return String.fromCharCode(parseInt(c,16));});
 		return raw_unescapexml(s.slice(0, i)) + s.slice(i+9,j) + raw_unescapexml(s.slice(j+3));
 	}
 	return function unescapexml(text/*:string*/, xlsx/*:boolean*/) {
@@ -16249,7 +16254,7 @@ return function parse_ws_xml_data(sdata/*:string*/, s, opts, guess/*:Range*/, th
 			d = x.slice(i);
 			p = ({t:""}/*:any*/);
 
-			if((cref=str_match_xml_ns(d, "v"))!= null && /*::cref != null && */cref[1] !== '') p.v=unescapexml(cref[1]);
+			if((cref=str_match_xml_ns(d, "v"))!= null && /*::cref != null && */cref[1] !== '') p.v = tag.t === 'str' ? cref[1] : unescapexml(cref[1]);
 			if(opts.cellFormula) {
 				if((cref=str_match_xml_ns(d, "f"))!= null /*:: && cref != null*/) {
 					if(cref[1] == "") {
